packages/polaris-viz/src/components/LineChart/utilities/getAlteredLineChartPosition.ts

```diff
@@ -0,0 +1,141 @@
+import type {BoundingRect, Dimensions} from '@shopify/polaris-viz-core';
+import {clamp} from '@shopify/polaris-viz-core';
+
+import type {TooltipPositionOffset} from '../../TooltipWrapper';
+import type {Margin} from '../../../types';
+import {TooltipHorizontalOffset} from '../../TooltipWrapper';
+
+// The space between the cursor and the tooltip
+export const TOOLTIP_MARGIN = 20;
+
+export interface AlteredPositionProps {
+  bandwidth: number;
+  chartBounds: BoundingRect;
+  currentX: number;
+  currentY: number;
+  isPerformanceImpacted: boolean;
+  margin: Margin;
+  position: TooltipPositionOffset;
+  tooltipDimensions: Dimensions;
+}
+
+export interface AlteredPositionReturn {
+  x: number;
+  y: number;
+}
+
+export type AlteredPosition = (
+  props: AlteredPositionProps,
+) => AlteredPositionReturn;
+
+export function getAlteredLineChartPosition(
+  props: AlteredPositionProps,
+): AlteredPositionReturn {
+  const {currentX, currentY, position, chartBounds} = props;
+
+  const newPosition = {...position};
+
+  let x = currentX;
+  let y = currentY;
+
+  //
+  // Y POSITIONING
+  //
+
+  if (props.isPerformanceImpacted) {
```

Comment:
Just curious, when does this prop become true?

Reply:
There's logic based on the amount of data points that will be rendered.